table/column.go

```diff
@@ -370,7 +371,13 @@ func GetZeroValue(col *model.ColumnInfo) types.Datum {
 		d.SetFloat64(0)
 	case mysql.TypeNewDecimal:
 		d.SetMysqlDecimal(new(types.MyDecimal))
-	case mysql.TypeString, mysql.TypeVarString, mysql.TypeVarchar:
+	case mysql.TypeString:
+		if col.Flen > 0 && col.Charset == charset.CharsetBin {
+			d.SetBytes(make([]byte, col.Flen))
+		} else {
+			d.SetString("")
+		}
+	case mysql.TypeVarString, mysql.TypeVarchar:
```

Comment:
should we consider `TypeVarString` ?

Reply:
No, `VarString` is not fixed length type, does not padding to the length.